### ISSUE TITLE
Skip OIDC 'openid' scope redirect parameter 

### ISSUE THREAD

**Issue by rafaelsousa** [quarkusio/quarkus#14093]
**Description**
When trying to create an application to connect to spotify oauth2 server their server answers :

Illegal scope

Looking at the URL generated by the quarkus extension, the parameter is passed as "&scope=openid+user-read-private+user-read-email"

This "OpenID" value is passed directly in the code, as seen in the quarkus openid extension :

https://github.com/quarkusio/quarkus/blob/068b19c036ac385c941708aa073f6e3e31d9f3ef/extensions/oidc/runtime/src/main/java/io/quarkus/oidc/runtime/CodeAuthenticationMechanism.java#L211

When removing the "openid" parameter, my application works, so it would be interesting to have an option for don't send it. 

**Implementation ideas**

As suggested for @sberyozkin, put a boolen property if it should be suppressed. 

https://quarkusio.zulipchat.com/#narrow/stream/187030-users/topic/oidc/near/221510904

**Comment by pedroigor** [comment]
Perhaps, only add the `oidc` scope if no scope was provided?

**Comment by rafaelsousa** [comment]
If the scope is [mandatory](https://openid.net/specs/openid-connect-core-1_0.html#AuthRequest) shouldn't the extension throw an error, informing the user to provide it in .properties file?

**Comment by sberyozkin** [comment]
Hi @pedroigor @rafaelsousa dropping it if the scopes have been provided would be a breaking change, and would also likely cause some headaches. Pedro, you are right, `openid` is required for OIDC, and since the extension is primarily targeting OIDC providers, most users who work with OIDC and need some extra scopes are already used to just adding `email`, etc, so I'd be a bit hesitant in asking them to start adding a required `oidc` just so that we can support pure OAuth2 providers.
Actually, this is an interesting point. @rafaelsousa can you confirm please `Spotify` providers returns `ID token` after the user authentication ?

**Comment by sberyozkin** [comment]
@pedroigor @rafaelsousa according to
https://developer.spotify.com/documentation/general/guides/authorization-guide/

ID token is not returned

so I'm not sure what are we going to do about this issue.

**Comment by rafaelsousa** [comment]
Hi @sberyozkin, I'm going to see in spotify side about this issue and I'll be back to this thread. Thank you.

**Comment by sberyozkin** [comment]
Hey @pedroigor It appears that if we want to support pure OAuth2 providers doing the authorization code flow, then it should be an independent extension - as in such cases it is mainly about the Quarkus endpoints getting the user authorization to propagate the access tokens somehow, it is the mute point if SSO can be assured without the ID token. Or we can add a token state strategy AT_RT only but I guess it would move us out of the OIDC code flow space completely :-). So may be an independent extension would be sounder...

**Comment by sberyozkin** [comment]
@rafaelsousa FYI, it will be possible to drop it once #24774 is merged. See also https://quarkus.io/guides/security-openid-connect-web-authentication#integration-with-github-and-other-oauth2-providers.

I'll open another issue to support Spotify confuration as `quarkus.oidc.provider=spotify`
